strax/context.py

```diff
@@ -1283,6 +1295,29 @@ def _check_forbidden(self):
         Otherwise, try to make it a tuple"""
         self.context_config['forbid_creation_of'] = strax.to_str_tuple(
             self.context_config['forbid_creation_of'])
+    
+    def _apply_function(self, data, targets):
+        """
+        Apply functions stored in the context config to any data that is returned via
+            get_array, get_df or accumulate. Functions stored in
+            context_config['apply_data_function'] should take exactly two positional
+            arguments: data and targets.
+        :param data: Any type of data
+        :param targets: list/tuple of strings of data type names to get
+        :return: the data after applying the function(s)
+        """
+        apply_functions = self.context_config['apply_data_function']
+        if not isinstance(apply_functions, (tuple, list)):
+            raise ValueError(f"apply_data_function in context config should be tuple of "
+                             f"functions. Instead got {apply_functions}")
+        for function in apply_functions:
+            if not hasattr(function, '__call__'):
+                raise TypeError(f'apply_data_function in the context_config got '
+                                f'{function} but expected callable function with two '
+                                f'positional arguments: f(data, targets).')
+            # Make sure that the function takes two arguments (data and targets)
+            data = function(data, targets)
```

Comment:
Maybe, just check how traceable a "targets" does not exist error would be. In case the error message is not traceable/understandable at all (e.g. due to numba), maybe add an explicit check with ValueError

Reply:
This is about as understandable as it gets:
![afbeelding](https://user-images.githubusercontent.com/22295914/90370084-1107c380-e06d-11ea-9b63-f57f7ac8b5b1.png)